core/src/filesystem/directory_listing.rs

```diff
@@ -25,23 +29,31 @@ pub struct DirectoryListing {
 pub struct DirectoryListingFile {
 	pub is_directory: bool,
 	pub name: String,
-	pub path: String,
+	pub path: PathBuf,
 }
 
 impl DirectoryListingFile {
-	pub fn new(is_directory: bool, name: &str, path: &str) -> DirectoryListingFile {
-		DirectoryListingFile {
+	pub fn new(is_directory: bool, name: &str, path: &str) -> Self {
+		Self {
 			is_directory,
 			name: name.to_string(),
-			path: path.to_string(),
+			path: Path::new(path).into(),
+		}
+	}
+
+	pub fn from_pathbuf(path: PathBuf) -> Self {
+		Self {
+			name: path.file_parts().file_name,
+			is_directory: path.is_dir(),
+			path,
 		}
 	}
```

Comment:
I think this is a good candidate for a `From` trait, e.g. `Impl From<PathBuf> for DirectoryListingFile`